internal/provider/datasources/block.go

```diff
@@ -0,0 +1,201 @@
+	resp.Schema = schema.Schema{
+		Description: `
+Get information about an existing Block by ID.
+<br>
+Use this data source to obtain Block-specific attributes, such as the data.
+`,
+		Attributes: map[string]schema.Attribute{
+			"id": schema.StringAttribute{
+				Required:    true,
+				CustomType:  customtypes.UUIDType{},
+				Description: "Block ID (UUID)",
+			},
+			"created": schema.StringAttribute{
+				Computed:    true,
+				CustomType:  customtypes.TimestampType{},
+				Description: "Timestamp of when the resource was created (RFC3339)",
+				Optional:    true,
+			},
+			"updated": schema.StringAttribute{
+				Computed:    true,
+				CustomType:  customtypes.TimestampType{},
+				Description: "Timestamp of when the resource was updated (RFC3339)",
+				Optional:    true,
+			},
+			"account_id": schema.StringAttribute{
+				CustomType:  customtypes.UUIDType{},
+				Description: "Account ID (UUID), defaults to the account set in the provider",
+				Optional:    true,
+			},
+			"workspace_id": schema.StringAttribute{
+				CustomType:  customtypes.UUIDType{},
+				Description: "Workspace ID (UUID), defaults to the workspace set in the provider",
+				Optional:    true,
+			},
+			"name": schema.StringAttribute{
+				Computed:    true,
+				Description: "Name of the block",
+				Optional:    true,
+			},
+			"data": schema.StringAttribute{
+				Computed:    true,
+				Sensitive:   true,
+				CustomType:  jsontypes.NormalizedType{},
+				Description: "The user-inputted Block payload, as a JSON string. The value's schema will depend on the selected `type` slug. Use `prefect block types inspect <slug>` to view the data schema for a given Block type.",
+				Optional:    true,
+			},
+			"block_schema_id": schema.StringAttribute{
+				Computed:    true,
+				Description: "Block schema ID (UUID)",
+				CustomType:  customtypes.UUIDType{},
+				Optional:    true,
+			},
+			"block_type_id": schema.StringAttribute{
+				Computed:    true,
+				Description: "Block type ID (UUID)",
+				CustomType:  customtypes.UUIDType{},
+				Optional:    true,
+			},
+			"block_type_name": schema.StringAttribute{
+				Computed:    true,
+				Description: "Block type name",
+				Optional:    true,
+			},
+		},
+	}
+}
+
+// Read refreshes the Terraform state with the latest data.
+func (d *blockDataSource) Read(ctx context.Context, req datasource.ReadRequest, resp *datasource.ReadResponse) {
+	var state BlockDataSourceModel
+
+	diag := req.Config.Get(ctx, &state)
+	resp.Diagnostics.Append(diag...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	client, err := d.client.BlockDocuments(state.AccountID.ValueUUID(), state.WorkspaceID.ValueUUID())
+	if err != nil {
+		resp.Diagnostics.AddError(
+			"Error creating block client",
+			fmt.Sprintf("Could not create block client, unexpected error: %s. This is a bug in the provider, please report this to the maintainers.", err.Error()),
+		)
```

Comment:
for client init errors, i have a helper called `CreateClientErrorDiagnostic` that you can use

Reply:
Ah right, thanks - updated in fc16a27. Was following along with the `Variable` example. We'll go back at some point and make use of that helper more consistently.

Reply:
for sure. as you can tell, i am an abomination to the consistency gods

Reply:
Hey you wrote the helper for consistency 😄